packages/flutter/test/widgets/sliver_main_axis_group_test.dart

```diff
@@ -795,7 +795,79 @@ void main() {
     );
     Scrollable.ensureVisible(key.currentContext!);
     await tester.pumpAndSettle();
-    expect(tester.getTopLeft(find.byKey(key)), Offset.zero);
+    // Regression test for https://github.com/flutter/flutter/issues/154615
+    expect(tester.getTopLeft(find.byKey(key)), const Offset(0, 300));
+  });
+
+  testWidgets('SliverMainAxisGroup scrolls to the correct position', (WidgetTester tester) async {
+    final ScrollController controller = ScrollController();
+    addTearDown(controller.dispose);
+    final FocusNode textFieldFocus = FocusNode();
+    addTearDown(textFieldFocus.dispose);
+    final FocusNode textFieldFocus2 = FocusNode();
+    addTearDown(textFieldFocus2.dispose);
+
+    await tester.pumpWidget(
+      _buildSliverMainAxisGroup(
+        controller: controller,
+        slivers: <Widget>[
+          SliverPersistentHeader(
+            pinned: true,
+            delegate: _SliverTitleDelegate(
+              child: Container(
+                color: Colors.red,
+                height: 60.0,
+              ),
+              height: 60.0,
+            ),
+          ),
+          SliverToBoxAdapter(
+              child: Material(
+            child: TextField(
+              focusNode: textFieldFocus,
+            ),
+          )),
+          SliverToBoxAdapter(
+            child: Container(
+              color: Colors.green,
+              height: 500,
+            ),
+          ),
+          SliverToBoxAdapter(
+            child: Container(
+              color: Colors.black,
+              height: 500,
+            ),
+          ),
+          SliverToBoxAdapter(
```

Comment:
Do we need this many containers/textfields? Is it possible to make the smallest use case for this test?